python/paddle/fluid/dygraph/dygraph_to_static/list_transformer.py

```diff
@@ -0,0 +1,169 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import gast
+import astor
+from paddle.fluid.dygraph.dygraph_to_static.static_analysis import AstNodeWrapper
+from .static_analysis import AstNodeWrapper, NodeVarType, StaticAnalysisVisitor
+
+
+class ListTransformer(gast.NodeTransformer):
+    """
+    This class transforms python list used in control flow into Static Graph Ast
+    """
+
+    def __init__(self, wrapper_root):
+        assert isinstance(
+            wrapper_root, AstNodeWrapper
+        ), "Input non-AstNodeWrapper node for the initialization of ListTransformer."
+        self.wrapper_root = wrapper_root
+        self.root = wrapper_root.node
+        self.name_of_list_set = set()
+        self.list_name_to_updated = dict()
+
+        self.static_analysis_visitor = StaticAnalysisVisitor(self.root)
+        self.node_to_wrapper_map = self.static_analysis_visitor.get_node_to_wrapper_map(
+        )
+        var_env = self.static_analysis_visitor.get_var_env()
+        var_env.cur_scope = var_env.cur_scope.sub_scopes[0]
+        self.scope_var_type_dict = var_env.get_scope_var_type()
+
+    def transform(self):
+        self.visit(self.root)
+        self.replace_list_with_tensor_array(self.root)
+
+    def visit_Assign(self, node):
+        if self._update_list_name_to_updated(node):
+            return node
+        return node
```

Comment:
The function `self._update_list_name_to_updated` doesn't change anything of `node`, you can refine the code of `return node` like:
```python
    def visit_Assign(self, node):
        self._update_list_name_to_updated(node)
        return node
```

Reply:
Yes. Thanks.